### PR TITLE
editoast:  split track section endpoint

### LINKED ISSUE #7007: back: split track in infra editor

### PR DESCRIPTION
**Pull request by sim51**
close #7007

### CODE REVIEW

File: editoast/src/views/infra/edition.rs [hunk 1]
```diff
@@ -54,6 +80,753 @@ pub async fn edit<'a>(
...
+            })
+            .collect_vec(),
+        curves: tracksection
+            .curves
+            .iter()
+            .filter(|e| e.end >= distance)
+            .map(|e| {
+                let mut item = e.clone();
+                if item.begin < distance {
+                    item.begin = distance;
```
Comment (flomonster) [inline editoast/src/views/infra/edition.rs]: This has to be fixed everywhere
```suggestion
                    item.begin = 0.;
```

File: editoast/src/views/infra/edition.rs [hunk 2]
```diff
@@ -54,6 +80,753 @@ pub async fn edit<'a>(
...
+        TrackEndpoint {
+            track: Identifier::from(left_tracksection_id),
+            endpoint: Endpoint::End,
+        },
+    );
+    ports.insert(
+        "B".into(),
+        TrackEndpoint {
+            track: Identifier::from(right_tracksection_id),
+            endpoint: Endpoint::End,
```
Comment (flomonster) [inline editoast/src/views/infra/edition.rs]:
```suggestion
            endpoint: Endpoint::Begin,
```

File: editoast/src/views/infra/edition.rs [hunk 3]
```diff
@@ -54,6 +80,753 @@ pub async fn edit<'a>(
...
+        right_tracksection_id,
+    ));
+
+    // last operation, we delete the given track
+    operations.push(Operation::Delete(DeleteOperation {
+        obj_type: ObjectType::TrackSection,
+        obj_id: payload.id.to_string(),
+    }));
+
+    println!("{:#?}", operations);
```
Comment (flomonster) [inline editoast/src/views/infra/edition.rs]:
```suggestion
```

File: editoast/src/views/infra/edition.rs [hunk 4]
```diff
@@ -54,6 +80,753 @@ pub async fn edit<'a>(
...
+    let mut operations: Vec<Operation> = [
+        Operation::Create(Box::new(RailjsonObject::TrackSection {
+            railjson: left_tracksection.into(),
+        })),
+        Operation::Create(Box::new(RailjsonObject::TrackSection {
+            railjson: right_tracksection.into(),
+        })),
+        Operation::Create(Box::new(RailjsonObject::Switch {
+            railjson: track_link.into(),
+        })),
```
Comment (flomonster) [inline editoast/src/views/infra/edition.rs]: These three operations could be calculated in a separate function.

**Review by flomonster — CHANGES_REQUESTED** [review]
We should add a bunch of tests.

File: editoast/src/views/infra/edition.rs (hunk 4 above)
Comment (sim51) [reply editoast/src/views/infra/edition.rs]: agree, but it's only used here and simple to understand in the flow of the function.
So IMHO, it's not needed.

**Review by flomonster — APPROVED** [review]
LGTM (reviewed + tested)

File: front/public/locales/en/errors.json [hunk 5]
```diff
@@ -49,7 +49,8 @@
     "infra": {
       "NotFound": "",
       "edition": {
-        "InfraIsLocked": "Infra is locked"
+        "InfraIsLocked": "Infra is locked",
+        "SplitTrackSectionBadOffset": "Distance to split trackSection '{{tracksection_id}}' in infra '{{infra_id}}' is invalid. It must be between 0 and {{tracksection_length}} meters"
```
Comment (SharglutDev) [inline front/public/locales/en/errors.json]: We should always use the full word even in english (a issue will be done for harmonization in the rest of the app)
```suggestion
        "InfraIsLocked": "Infrastructure is locked",
        "SplitTrackSectionBadOffset": "Distance to split trackSection '{{tracksection_id}}' in infrastructure '{{infra_id}}' is invalid. It must be between 0 and {{tracksection_length}} meters".
```

File: front/public/locales/fr/errors.json [hunk 6]
```diff
@@ -49,7 +49,8 @@
     "infra": {
       "NotFound": "",
       "edition": {
-        "InfraIsLocked": "Infrastructure verrouillée"
+        "InfraIsLocked": "Infrastructure verrouillée",
+        "SplitTrackSectionBadOffset": "La distance pour scinder la section de ligne '{{tracksection_id}}' de l'infra '{{infra_id}}' est invalide. La valeur doit être comprise entre 0 et {{tracksection_length}} mètres"
```
Comment (SharglutDev) [inline front/public/locales/fr/errors.json]:
```suggestion
        "SplitTrackSectionBadOffset": "La distance pour scinder la section de ligne '{{tracksection_id}}' de l'infrastructure '{{infra_id}}' est invalide. La valeur doit être comprise entre 0 et {{tracksection_length}} mètres."
```

**Review by SharglutDev — COMMENTED** [review]
Thank you for your PR ! Just 2 comments for trads :)

File: front/public/locales/fr/errors.json [hunk 7]
```diff
@@ -184,7 +185,7 @@
     "train_schedule_v2": {
       "BatchTrainScheduleNotFound": "'{{number}}' circulation(s) n'ont pas pu être trouvée(s)",
       "NotFound": "Circulation '{{train_schedule_id}}' non trouvée",
-      "InfraNotFound": "Infra '{{infra_id}}' non trouvée"
+      "InfraNotFound": "infrastructure '{{infra_id}}' non trouvée"
```
Comment (SharglutDev) [inline front/public/locales/fr/errors.json]:
```suggestion
      "InfraNotFound": "Infrastructure '{{infra_id}}' non trouvée"
```

File: front/public/locales/en/errors.json [hunk 8]
```diff
@@ -164,8 +165,8 @@
       "StartDateAfterEndDate": "The study start date must be before the end date"
     },
     "timetable": {
-      "InfraNotLoaded": "Infra '{{infra_id}}' is not loaded",
-      "InfraNotFound": "Infra {{infra_id}} does not exist",
+      "InfraNotLoaded": "Infrastructure '{{infra_id}}' is not loaded",
+      "InfraNotFound": "Infrastructure {{infra_id}} does not exist",
```
Comment (SharglutDev) [inline front/public/locales/en/errors.json]:
```suggestion
      "InfraNotFound": "Infrastructure '{{infra_id}}' does not exist",
```

File: front/public/locales/en/errors.json [hunk 9]
```diff
@@ -151,7 +152,7 @@
       "UnknownSignalingSystem": "Unknown signaling system"
     },
     "stdcm": {
-      "InfraNotFound": "Infra {{infra_id}} does not exist"
+      "InfraNotFound": "Infrastructure {{infra_id}} does not exist"
```
Comment (SharglutDev) [inline front/public/locales/en/errors.json]:
```suggestion
      "InfraNotFound": "Infrastructure '{{infra_id}}' does not exist"
```

File: front/public/locales/en/errors.json [hunk 10]
```diff
@@ -87,7 +88,7 @@
     "pathfinding": {
       "ElectricalProfilesOverlap": "Electrical Profile overlaps with others",
       "ElectrificationOverlap": "Electrification {{electrification_id}} overlaps with other electrifications",
-      "InfraNotFound": "Infra {{infra_id}} does not exist",
+      "InfraNotFound": "Infrastructure {{infra_id}} does not exist",
```
Comment (SharglutDev) [inline front/public/locales/en/errors.json]:
```suggestion
      "InfraNotFound": "Infrastructure '{{infra_id}}' does not exist",
```

File: front/public/locales/en/errors.json [hunk 11]
```diff
@@ -49,7 +49,8 @@
     "infra": {
       "NotFound": "",
       "edition": {
-        "InfraIsLocked": "Infra is locked"
+        "InfraIsLocked": "Infrastructure is locked",
+        "SplitTrackSectionBadOffset": "Distance to split trackSection '{{tracksection_id}}' in infrastructure '{{infra_id}}' is invalid. It must be between 0 and {{tracksection_length}} meters."
```
Comment (SharglutDev) [inline front/public/locales/en/errors.json]:
```suggestion
        "SplitTrackSectionBadOffset": "Distance to split track section '{{tracksection_id}}' in infrastructure '{{infra_id}}' is invalid. It must be between 0 and {{tracksection_length}} meters."
```

**Review by SharglutDev — APPROVED** [review]
Lgtm !

**Pull request merged**